### PR TITLE
rqt_bag : Addition of a step-by-step play

### PR DESCRIPTION
**Pull request by suryaambrose**
rqt_bag : Addition of buttons to allow a message-by-message navigation in the bag

### CODE REVIEW

File: rqt_bag/src/rqt_bag/bag_timeline.py [hunk 1]
```diff
@@ -752,12 +781,26 @@ def toggle_play(self):
...
+        self.last_playhead = self._timeline_frame.playhead
         self._play_timer.start()
 
     def navigate_stop(self):
         self.play_speed = 0.0
         self._play_timer.stop()
 
+    def navigate_previous(self):
+        self.navigate_stop()
+        self.play_speed = -1.0
```
Comment (ablasdel) [inline rqt_bag/src/rqt_bag/bag_timeline.py]: Why are you overriding the navigate_stop play_speed here?

If the button is intended to step through the timesteps the play_speed should be 0.0 after pressed. 
I think this line should simply be removed. I tested it without it and it still works.

File: rqt_bag/src/rqt_bag/bag_timeline.py [hunk 2]
```diff
@@ -752,12 +781,26 @@ def toggle_play(self):
...
+        self._timeline_frame.playhead = self.get_previous_message_time()
+        self.last_playhead = self._timeline_frame.playhead
+
+    def navigate_next(self):
+        self.navigate_stop()
+        self.play_speed = 1.0
```
Comment (ablasdel) [inline rqt_bag/src/rqt_bag/bag_timeline.py]: same as above

**Comment by ablasdel** [comment]
This is a great addition! Thanks @suryaambrose, with those minor edits I will be able to merge this and it should make it into the jade release.

**Comment by ablasdel** [comment]
@suryaambrose I opened this PR https://github.com/suryaambrose/rqt_common_plugins/pull/1 

When you merge that PR I can merge this one.

**Comment by suryaambrose** [comment]
Done sorry for the lag

**Comment by ablasdel** [comment]
No problem thanks for the PR!

**Pull request merged**